core/src/apps/cardano/sign_tx/signer.py

```diff
@@ -458,6 +616,62 @@ def _validate_token(
         if len(token.asset_name_bytes) > MAX_ASSET_NAME_LENGTH:
             raise INVALID_TOKEN_BUNDLE
 
+    # inline datum
+
+    async def _process_inline_datum(
```

Comment:
I tried to extract the chunk processing into a separate function in a separate branch https://github.com/vacuumlabs/trezor-firmware/commit/85f0e79c020bac7a3fff91a034fa840b2ecd5c6f. I'm 50/50 on whether it's better or not. What do you think?

Reply:
I agree that the current implementation contains some duplication, but I'm not sure whether introducing multiple if-then-else statements is much better:

```python
title = "Inline datum" if chunk_type == "inline-datum" else "Reference script"
```
```python
br_type="confirm_inline_datum"
if chunk_type == "inline-datum"
else "confirm_reference_script"
```
```python
message_type = (
    messages.CardanoTxInlineDatumChunk
    if chunk_type == "inline-datum"
    else messages.CardanoTxReferenceScriptChunk
)
```

Personally, I slightly prefer the original implementation (I find it similarly "ugly", but a bit easier to reason about), but if you conclude that the extracted version is better, I'll be okay with that as well. Alternatively, we could ask Trezor devs what they think when opening the PR.